CoffeeShop.playground/Contents.swift

```diff
@@ -0,0 +1,413 @@
+import Foundation
+
+// MARK: - Location
+struct Location: Equatable {
+    var x = 0.0
+    var y = 0.0
+
+    func distanceTo(_ location: Location) -> Double {
+        return sqrt( pow(self.x - location.x, 2) + pow(self.y - location.y, 2) )
+    }
+}
+
+
+typealias CoffeeShopWithDistance = (coffeeShop: CoffeeShop, distance: Double)
+
+// MARK: - CoffeeShop
+struct CoffeeShop: Equatable {
+
+    var name: String
+    var location: Location
+
+    init(name: String, location: Location) {
+        self.name = name
+        self.location = location
+    }
+
+    init?(_ line: String, separator: String) {
+        let components = line.components(separatedBy: separator)
+        guard !separator.isEmpty,
+            components.count == 3,
+            !components[0].isEmpty,
+            let x = Double(components[1]),
+            let y = Double(components[2])
+            else {
+                return nil
+        }
+
+        self.init(name: components[0], location: Location(x: x, y: y))
+    }
+
+    static func coffeeShops(fromCSV content: String) -> [CoffeeShop]? {
+        var coffeeShops = [CoffeeShop]()
+        let lines = content.components(separatedBy: .newlines)
+        for line in lines {
+            guard let coffeeShop = CoffeeShop(line, separator: ",") else {
+                return nil
+            }
+            coffeeShops.append(coffeeShop)
+        }
+        return coffeeShops
+    }
+
+    static func sort(coffeeShops: [CoffeeShop], byDistanceTo location: Location, ascending: Bool = true) -> [CoffeeShopWithDistance] {
+        var coffeeShopWithDistance = coffeeShops.map { CoffeeShopWithDistance(coffeeShop: $0, distance: $0.location.distanceTo(location)) }
+        coffeeShopWithDistance.sort {
+            if ascending {
+                return $0.distance < $1.distance
+            } else {
+                return $0.distance > $1.distance
+            }
+        }
+        return coffeeShopWithDistance
+    }
+}
+
+// MARK: - CoffeeShopApp
+struct CoffeeShopApp {
```

Comment:
Prefer to have 1 class / file. It would be much easier to navigate.

Reply:
I'll refactor this